contracts/access/AccessControlAdminRules.sol

```diff
@@ -0,0 +1,228 @@
+    }
+
+    /**
+     * @dev See {IAccessControlAdminRules-acceptAdminTransfer}
+     */
+    function acceptAdminTransfer() public virtual {
+        address pendingAdminOwner = pendingAdmin();
+        require(
+            _hasAdminTransferDelayPassed() && _msgSender() == pendingAdminOwner,
+            "AccessControl: delay must be met and caller must be pending admin"
+        );
+        _revokeRole(DEFAULT_ADMIN_ROLE, admin());
+        _grantRole(DEFAULT_ADMIN_ROLE, pendingAdminOwner);
+        _resetAdminTransfer();
+    }
+
+    /**
+     * @dev See {IAccessControlAdminRules-cancelAdminTransfer}
+     */
+    function cancelAdminTransfer() public virtual onlyRole(DEFAULT_ADMIN_ROLE) {
+        _resetAdminTransfer();
+    }
+
+    /**
+     * @dev Revokes `role` from the calling account.
+     *
+     * For `DEFAULT_ADMIN_ROLE`, only allows renouncing in two steps, so it's required
+     * that the {delayedUntil} is met and the pending admin is the zero address.
+     * After its execution, it will not be possible to call `onlyRole(DEFAULT_ADMIN_ROLE)`
+     * functions.
+     *
+     * For other roles, see {AccessControl-renounceRole}.
+     *
+     * NOTE: Renouncing `DEFAULT_ADMIN_ROLE` will leave the contract without an owner,
+     * thereby disabling any functionality that is only available to the default admin, and the
+     * possibility of reassigning a non-administrated role.
+     */
+    function renounceRole(bytes32 role, address account) public virtual override(IAccessControl, AccessControl) {
+        if (role == DEFAULT_ADMIN_ROLE) {
+            require(
+                pendingAdmin() == address(0) && _hasAdminTransferDelayPassed(),
+                "AccessControl: admin can only renounce in two delayed steps"
+            );
+        }
+        super.renounceRole(role, account);
+    }
+
+    /**
+     * @dev See {AccessControl-grantRole}. Reverts for `DEFAULT_ADMIN_ROLE`.
+     */
+    function grantRole(
+        bytes32 role,
+        address account
+    ) public virtual override(IAccessControl, AccessControl) onlyRole(getRoleAdmin(role)) {
+        require(role != DEFAULT_ADMIN_ROLE, "AccessControl: can't directly grant admin role");
+        super.grantRole(role, account);
+    }
+
+    /**
+     * @dev See {AccessControl-revokeRole}. Reverts for `DEFAULT_ADMIN_ROLE`.
+     */
+    function revokeRole(
+        bytes32 role,
+        address account
+    ) public virtual override(IAccessControl, AccessControl) onlyRole(getRoleAdmin(role)) {
+        require(role != DEFAULT_ADMIN_ROLE, "AccessControl: can't directly revoke admin role");
+        super.revokeRole(role, account);
+    }
+
+    /**
+     * @dev See {AccessControl-_setRoleAdmin}. Reverts for `DEFAULT_ADMIN_ROLE`.
+     */
+    function _setRoleAdmin(bytes32 role, bytes32 adminRole) internal virtual override {
+        require(role != DEFAULT_ADMIN_ROLE, "AccessControl: can't override admin's admin");
```

Comment:
>admin's admin

See here it is very clear that we have a name clash between two different concepts.